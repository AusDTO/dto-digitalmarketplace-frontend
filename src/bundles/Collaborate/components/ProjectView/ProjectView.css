main .sidebar {
  float: left;
  margin: 0 2.04556% 0 0;
}

.actions {
  text-align: right;
}

.actions a {
  display: inline-block;
}
.actions a:first-of-type {
  margin-right: 1em;
}
.stage{
  margin-right: 14px;
  margin-bottom: 10px;
  padding: 3px 6px 3px 6px;
  font-size: 14px;
  border-radius: 2px;
  border: solid 1px #17788d;
  float: left;
}
.full-header {
  background: #f0f3f5;
  padding-top: 1.5em;
}

.full-header h4:first-of-type, .full-header ul {
  margin-top: 0;
}

.full-header > div {
  max-width: 75em;
  margin: 0 auto;
  padding: 0 1.6em;
}
.project ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

.tile {
  border: solid 1px rgba(23, 120, 141, 0.5);
  box-shadow: 0 0 5px 0 rgba(187, 187, 187, 0.5);
  padding: 1.5rem 2rem;
  border-radius: 4px;
  text-align: center;
  margin-bottom: 2em;
  background-color: white;
}
.tile .title {
   display: block;
   color: #6e6e6e;
   margin-bottom: 10px;
   font-size: 20px;
 }

.tile b {
  display: block;
  margin-bottom: 20px;
  font-size: 20px;
}

.tile p {
  margin-bottom: 20px;
  font-size: 17px;
}

.tile a[role="button"] {
  color: white;
  width: 100%;
  height: auto;
  margin: 0;
  border-radius: 4px;
}

.center {
  max-width: 75em;
  margin: 0 auto;
  padding: 0 1.6em;
}

.summary {
  font-size: 20px;
}