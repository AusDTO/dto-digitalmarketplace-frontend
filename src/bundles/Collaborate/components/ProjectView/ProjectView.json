{
  "project": {
    "id": 2,
    "title": "Finding value in asset data",
    "opportunity": "Council owns assets across the City of Casey that are provided for the benefit of the community. These assets include community facilities, roads, buildings, trees, parks, drains, sports grounds, play grounds and nature reserves. These assets require an ongoing commitment from Council to manage their operations and to maintain them, or to renew, upgrade or expand the assets when they reach the end of their useful life. Council is committed (per the Asset Management Policy) to looking after these assets in a way that meets the expectations of present and future residents, at the lowest economic and environmental cost.\n\nCouncil maintains several datasets that contain information about these assets. Ideally, these datasets would be used to draw insights to inform long term strategic and budget decisions. We believe there are opportunities to link up our datasets and apply statistical & predictive modelling to save money and improve asset management.",
    "client": "City of Casey",
    "contact_name": "Casey",
    "contact_email": "casey@casey.vic.gov.au",
    "service": "Planning, Smart Cities",
    "stage": "Discovery",
    "timeframe": "January 2016 — June 2016",
    "problem": "Asset data is stored across a range of disparate and inconsistently structured datasets. \nEach dataset is maintained to different standards. For instance, the parks and reserves dataset is regularly  updated by dedicated resources. The bus shelters dataset hasn’t been updated in the last three years.\nThe inconsistent nature of these datasets means they cannot be used to reliably inform long term budget or strategic planning decisions. We think the most valuable datasets (ie. relating to those assets where Council invests the most) are:\n-        Civil Assets (for example roads, paths, drains),\n-        Buildings, and\n-        Sports grounds.\nBecause our maturity in analysing data is low, we need some help to identify and capitalise on opportunities.",
    "users": "Given the purpose of this project is to run a discovery, we do not yet have a backlog of researched and validated user stories. Instead, we have a hypothesis that we can use our data to make better strategic and tactical decisions that save money and make the best use of our limited Council resources.",
    "outcome": [
      "We're seeking a data science expert sourced from the Digital Marketplace for 8 weeks with a budget range of $50-75k. During this short term project will run a discovery process to assess the quality of our data and identify opportunities to make better use of our limited resources. "
    ],
    "project_links": [
      "https://marketplace.service.gov.au/search/sellers/?role=Data science"
    ],
    "meta": {
      "deleteLink": "#",
      "editLink": "#"
    }
  }
}