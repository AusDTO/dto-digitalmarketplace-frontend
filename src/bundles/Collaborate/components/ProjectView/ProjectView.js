import React, { PropTypes } from 'react';
import { connect } from 'react-redux';

import { newline } from '../../../../helpers';

import './ProjectView.css'

class ProjectView extends React.Component {
  state = { showConfirm: false }

  toggleConfirm(show = true) {
    this.setState({
      showConfirm: show
    }, () => this.state.showConfirm && this.refs.confirm.focus());
  }

  render() {
    const {
      title,
      opportunity,
      client,
      referee_name,
      referee_contact,
      supplier_name,
      supplier_url = null,
      approach,
      timeframe,
      outcome,
      project_links,
      service,
      stage,
      roles,
      meta,
      confirmButton = null,
      returnLink = null
    } = this.props;

    const { showConfirm } = this.state;

    return (
      <section id="project__ProjectView">
        {showConfirm && (
          <div ref="confirm" className="callout--warn" aria-labelledby="callout--success__heading" tabIndex="-1" role="alert">
              <p id="callout--success__heading">Are you sure you want to delete this case study?</p>
              <a href={meta.deleteLink} role="button">Delete this case study</a>
              <button className="button-secondary" onClick={this.toggleConfirm.bind(this, false)}>No, keep this case study</button>
          </div>
        )}
        <header className="row">
          <div className="col-xs-12">
            <h1 tabIndex="-1">{title}</h1>
          </div>

          <div className="meta col-xs-12">
            <div className="row">
              <div className="col-xs-12 col-sm-7">
                <p>by {client}</p>
              </div>
              {meta && false && (
                <div className="col-xs-12 col-sm-5 actions">
                  <a href={meta.editLink}>Edit project</a>
                  <button className="button-secondary" onClick={this.toggleConfirm.bind(this)}>Delete</button>
                </div>
              )}

            </div>
          </div>

        </header>
        <div className="row">

          <article role="main" className="col-sm-9 col-xs-12">

            <section>
              <h2>Council service</h2>
              <p className="freetext">{service}</p>
            </section>
            <section>
              <h2>Project Stage</h2>
              <p className="freetext">{stage}</p>
            </section>

            <section>
              <h2>Aim</h2>
              <p className="freetext">{newline(opportunity)}</p>
            </section>
            <section>
              <h2>How it ran</h2>
              <p className="freetext">{newline(approach)}</p>
            </section>

            <section>
              <h2>Outcomes and benefits</h2>
              <ul>
                {outcome.map((content, i) => <li key={i}>{content}</li>)}
              </ul>
            </section>
            {project_links && project_links.length > 0 && (
              <section className="project">
                <h2>Project Links</h2>
                <ul>
                  {project_links.map((item, i) => <li key={i}>
                    <a className="project__links" href={item} rel="external" target="_blank">{item}</a>
                  </li>)}
                </ul>
              </section>
            )}
            <div className="project__actions">
              {confirmButton}
              {returnLink}
            </div>
          </article>
          <aside className="col-sm-3 col-xs-12">


            <div styleName="tile">
              <span>{client}</span>
              <b>{referee_name}</b>

              <a href={`mailto:${referee_contact}`} role="button">Email organiser</a>
            </div>



          </aside>
        </div>
      </section>
    )
  }
}

ProjectView.propTypes = {
  title: PropTypes.string.isRequired,
  opportunity: PropTypes.string.isRequired,
  client: PropTypes.string.isRequired,
    referee_name: PropTypes.string.isRequired,
    referee_contact: PropTypes.string.isRequired,
  stage: PropTypes.string.isRequired,
  service: PropTypes.string.isRequired,
  approach: PropTypes.string.isRequired,
  timeframe: PropTypes.string.isRequired,
  outcome: PropTypes.arrayOf(PropTypes.string).isRequired,
  project_links: PropTypes.arrayOf(PropTypes.string),
  meta: PropTypes.objectOf(PropTypes.string),

  returnLink: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
}

const mapStateToProps = (state, ownProps) => {
  return {
    ...state.project,
    ...ownProps
  }
}

export default connect(mapStateToProps)(ProjectView)
