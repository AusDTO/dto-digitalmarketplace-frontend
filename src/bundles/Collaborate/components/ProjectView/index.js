import ProjectView from './ProjectView'
import sampleState from './ProjectView.json'

export { sampleState }
export default ProjectView
