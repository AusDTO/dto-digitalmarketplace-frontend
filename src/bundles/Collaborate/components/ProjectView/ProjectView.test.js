// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import ProjectView from './ProjectView';
import renderer from 'react-test-renderer';

import sampleState from './ProjectView.json';
import createStore from '../../redux/create';

test('ProjectView with required attributes', () => {
  const store = createStore(Object.assign({}, { _serverContext: {} },  sampleState));
  const component = renderer.create(
    <ProjectView store={store} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('ProjectView with required attributes and meta links', () => {
  let state = Object.assign({}, sampleState);
  state.project.meta = {
    editLink: 'link/to/edit',
    deleteLink: 'link/to/delete'
  };

  const store = createStore(Object.assign({}, { _serverContext: {} },  state));
  const component = renderer.create(
    <ProjectView store={store} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
