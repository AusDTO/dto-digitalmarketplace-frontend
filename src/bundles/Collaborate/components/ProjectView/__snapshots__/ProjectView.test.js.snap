// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectView with required attributes 1`] = `
<div>
  <section
    styleName="full-header"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <h1
          tabIndex="-1"
        >
          Finding value in asset data
        </h1>
      </div>
    </div>
    <div
      className="row"
    >
      <article
        className="col-xs-12 col-md-8 col-sm-7"
      >
        <p
          styleName="summary"
        />
        <p>
          <strong>
            Project stage
          </strong>
           
          Discovery
        </p>
      </article>
      <article
        className="col-xs-12 col-sm-8 col-sm-push-1 col-md-3 col-md-push-1"
      >
        <div
          styleName="tile"
        >
          <span>
            Council
          </span>
          <b>
            City of Casey
          </b>
          <a
            className="button"
            href="mailto:casey@casey.vic.gov.au"
          >
            Email contact
          </a>
        </div>
      </article>
    </div>
  </section>
  <section
    id="project__ProjectView"
    styleName="center"
  >
    <div
      className="row"
    >
      <aside
        className="col-sm-3 col-xs-12"
      />
      <article
        className="col-sm-7 col-xs-12"
        role="main"
      >
        <section>
          <h2
            className="au-display-lg"
          >
            What's the key problem to solve?
          </h2>
          <p
            className="freetext"
          >
            <span>
              Asset data is stored across a range of disparate and inconsistently structured datasets. 
              <br />
            </span>
            <span>
              Each dataset is maintained to different standards. For instance, the parks and reserves dataset is regularly  updated by dedicated resources. The bus shelters dataset hasn’t been updated in the last three years.
              <br />
            </span>
            <span>
              The inconsistent nature of these datasets means they cannot be used to reliably inform long term budget or strategic planning decisions. We think the most valuable datasets (ie. relating to those assets where Council invests the most) are:
              <br />
            </span>
            <span>
              -        Civil Assets (for example roads, paths, drains),
              <br />
            </span>
            <span>
              -        Buildings, and
              <br />
            </span>
            <span>
              -        Sports grounds.
              <br />
            </span>
            <span>
              Because our maturity in analysing data is low, we need some help to identify and capitalise on opportunities.
              <br />
            </span>
          </p>
        </section>
        <section>
          <h2
            className="au-display-lg"
          >
            Who are the users and their needs?
          </h2>
          <p
            className="freetext"
          >
            <span>
              Given the purpose of this project is to run a discovery, we do not yet have a backlog of researched and validated user stories. Instead, we have a hypothesis that we can use our data to make better strategic and tactical decisions that save money and make the best use of our limited Council resources.
              <br />
            </span>
          </p>
        </section>
        <section>
          <h2
            className="au-display-lg"
          >
            Outcomes
          </h2>
          <ul>
            <li>
              We're seeking a data science expert sourced from the Digital Marketplace for 8 weeks with a budget range of $50-75k. During this short term project will run a discovery process to assess the quality of our data and identify opportunities to make better use of our limited resources. 
            </li>
          </ul>
        </section>
        <section
          className="project"
        >
          <h2
            className="au-display-lg"
          >
            Project Links
          </h2>
          <ul>
            <li>
              <a
                className="project__links"
                href="https://marketplace.service.gov.au/search/sellers/?role=Data science"
                rel="external"
                target="_blank"
              >
                https://marketplace.service.gov.au/search/sellers/?role=Data science
              </a>
            </li>
          </ul>
        </section>
        <div
          className="project__actions"
        />
      </article>
    </div>
  </section>
</div>
`;

exports[`ProjectView with required attributes and meta links 1`] = `
<div>
  <section
    styleName="full-header"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <h1
          tabIndex="-1"
        >
          Finding value in asset data
        </h1>
      </div>
    </div>
    <div
      className="row"
    >
      <article
        className="col-xs-12 col-md-8 col-sm-7"
      >
        <p
          styleName="summary"
        />
        <p>
          <strong>
            Project stage
          </strong>
           
          Discovery
        </p>
      </article>
      <article
        className="col-xs-12 col-sm-8 col-sm-push-1 col-md-3 col-md-push-1"
      >
        <div
          styleName="tile"
        >
          <span>
            Council
          </span>
          <b>
            City of Casey
          </b>
          <a
            className="button"
            href="mailto:casey@casey.vic.gov.au"
          >
            Email contact
          </a>
        </div>
      </article>
    </div>
  </section>
  <section
    id="project__ProjectView"
    styleName="center"
  >
    <div
      className="row"
    >
      <aside
        className="col-sm-3 col-xs-12"
      />
      <article
        className="col-sm-7 col-xs-12"
        role="main"
      >
        <section>
          <h2
            className="au-display-lg"
          >
            What's the key problem to solve?
          </h2>
          <p
            className="freetext"
          >
            <span>
              Asset data is stored across a range of disparate and inconsistently structured datasets. 
              <br />
            </span>
            <span>
              Each dataset is maintained to different standards. For instance, the parks and reserves dataset is regularly  updated by dedicated resources. The bus shelters dataset hasn’t been updated in the last three years.
              <br />
            </span>
            <span>
              The inconsistent nature of these datasets means they cannot be used to reliably inform long term budget or strategic planning decisions. We think the most valuable datasets (ie. relating to those assets where Council invests the most) are:
              <br />
            </span>
            <span>
              -        Civil Assets (for example roads, paths, drains),
              <br />
            </span>
            <span>
              -        Buildings, and
              <br />
            </span>
            <span>
              -        Sports grounds.
              <br />
            </span>
            <span>
              Because our maturity in analysing data is low, we need some help to identify and capitalise on opportunities.
              <br />
            </span>
          </p>
        </section>
        <section>
          <h2
            className="au-display-lg"
          >
            Who are the users and their needs?
          </h2>
          <p
            className="freetext"
          >
            <span>
              Given the purpose of this project is to run a discovery, we do not yet have a backlog of researched and validated user stories. Instead, we have a hypothesis that we can use our data to make better strategic and tactical decisions that save money and make the best use of our limited Council resources.
              <br />
            </span>
          </p>
        </section>
        <section>
          <h2
            className="au-display-lg"
          >
            Outcomes
          </h2>
          <ul>
            <li>
              We're seeking a data science expert sourced from the Digital Marketplace for 8 weeks with a budget range of $50-75k. During this short term project will run a discovery process to assess the quality of our data and identify opportunities to make better use of our limited resources. 
            </li>
          </ul>
        </section>
        <section
          className="project"
        >
          <h2
            className="au-display-lg"
          >
            Project Links
          </h2>
          <ul>
            <li>
              <a
                className="project__links"
                href="https://marketplace.service.gov.au/search/sellers/?role=Data science"
                rel="external"
                target="_blank"
              >
                https://marketplace.service.gov.au/search/sellers/?role=Data science
              </a>
            </li>
          </ul>
        </section>
        <div
          className="project__actions"
        />
      </article>
    </div>
  </section>
</div>
`;
